ui/src/pages/users.vue

```diff
@@ -236,6 +251,58 @@
 
   <!-- Log in as User modal -->
   <SudoUserModal ref="sudoModal" :user="selected" />
+
+    <!-- Delete Confirmation Modal -->
+  <va-modal
+    v-model="isDeleteModalVisible"
+    title="DELETE USER?"
+    okText="Confirm"
+    cancelText="Cancel"
+    @ok="showUsernamePrompt"
+    @cancel="isDeleteModalVisible = false"
+  >
+    <p>Are you sure you want to delete the user record of <strong>{{ editedUser.name }}</strong> completely?</p>
+  </va-modal>
+
+  <!-- Username Confirmation Modal -->
+  <va-modal 
+    v-model="isUsernamePromptVisible"
+    title="CONFIRM DELETION"
+    hide-default-actions
+    @cancel="resetDeleteModal"
+  >
+    <p>This action is irreversible and will impact other data linked to this user.</p>
+    <p>Please type the username <strong>{{ editedUser.username }}</strong> to confirm deletion:</p>
+    <va-input v-model="usernameConfirmation" placeholder="Enter username" />
+
+    <!-- Radio buttons for delete cascade or update cascade -->
+    <div class="mt-4">
+      <label>
+        <input type="checkbox" :checked="deleteOption === 'hardDelete'" @change="toggleOption('hardDelete')" />
+        Hard Deletion
+      </label>
+      <label class="ml-4">
+        <input type="checkbox" :checked="deleteOption === 'softDelete'" @change="toggleOption('softDelete')" />
+        Soft Deletion
```

Comment:
Could you not show the `Soft Deletion` radio button option if the user is already soft-deleted? Instead, just show `Hard Deletion` and make it selected by default